lib/ae_mdw/db/util.ex

```diff
@@ -158,23 +166,21 @@ defmodule AeMdw.Db.Util do
 
   @spec key_block_height(state(), binary()) :: {:ok, Blocks.height()} | {:error, Error.t()}
   def key_block_height(state, hash_or_kbi) do
-    case Util.parse_int(hash_or_kbi) do
-      {:ok, kbi} when kbi >= 0 ->
-        last_gen = last_gen(state)
-
-        if kbi <= last_gen do
-          {:ok, kbi}
-        else
-          {:error, ErrInput.NotFound.exception(value: hash_or_kbi)}
-        end
-
-      {:ok, _kbi} ->
+    with {:ok, kbi} when kbi >= 0 <- Util.parse_int(hash_or_kbi),
+         {:ok, last_gen} <- last_gen(state) do
+      if kbi <= last_gen do
```

Comment:
I think it will be better if this is moved out in the `with` part, either as-is (the false will match with the `_invalid_kbi` case, or with wrapping it in a tuple with some descriptive atom.

Reply:
Good call, done